src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -205,45 +209,116 @@ private String convertIsoCountryCodeToFipsCountryCode(String isoCountryCode, Map
         allVendorAddresses.add(buildTaxAddress(pmwVendor, paymentWorksIsoToFipsCountryMap));
         allVendorAddresses.add(buildRemitAddress(pmwVendor, paymentWorksIsoToFipsCountryMap));
+        if (paymentWorksVendorIsPurchaseOrderVendor(pmwVendor)) {
+            allVendorAddresses.add(buildPurchaseOrderAddress(pmwVendor));
+        }
         return allVendorAddresses;
     }
 
     private VendorAddress buildTaxAddress(PaymentWorksVendor pmwVendor, Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
         VendorAddress taxAddress = buildBaseAddress(CUVendorConstants.CUAddressTypes.TAX, 
                                                     pmwVendor.getCorpAddressStreet1(),pmwVendor.getCorpAddressStreet2(),
                                                     pmwVendor.getCorpAddressCity(), pmwVendor.getCorpAddressZipCode(), 
-                                                    pmwVendor.getCorpAddressCountry(), paymentWorksIsoToFipsCountryMap);
-        taxAddress = assignStateOrProvinceBasedOnCountryCode(taxAddress, pmwVendor.getCorpAddressState());
+                                                    convertIsoCountryCodeToFipsCountryCode(pmwVendor.getCorpAddressCountry(), paymentWorksIsoToFipsCountryMap));
+        taxAddress = assignStateCodeOrProvinceBasedOnFipsCountryCode(taxAddress, pmwVendor.getCorpAddressState());
         return taxAddress;
     }
 
     private VendorAddress buildRemitAddress(PaymentWorksVendor pmwVendor, Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
         VendorAddress remitAddress = buildBaseAddress(VendorConstants.AddressTypes.REMIT,
                                                       pmwVendor.getRemittanceAddressStreet1(), pmwVendor.getRemittanceAddressStreet2(), 
                                                       pmwVendor.getRemittanceAddressCity(), pmwVendor.getRemittanceAddressZipCode(), 
-                                                      pmwVendor.getRemittanceAddressCountry(), paymentWorksIsoToFipsCountryMap);
+                                                      convertIsoCountryCodeToFipsCountryCode(pmwVendor.getRemittanceAddressCountry(), paymentWorksIsoToFipsCountryMap));
         remitAddress.setVendorDefaultAddressIndicator(true);
-        remitAddress = assignStateOrProvinceBasedOnCountryCode(remitAddress, pmwVendor.getRemittanceAddressState());
+        remitAddress = assignStateCodeOrProvinceBasedOnFipsCountryCode(remitAddress, pmwVendor.getRemittanceAddressState());
         return (remitAddress);
     }
 
-    private VendorAddress buildBaseAddress(String addressType, String line1, String line2, String city, String zip, String isoCountryCode, Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
+    private String convertFipsPoCountryOptionToFipsCountryCode(String poCountryOption) {
+        if (StringUtils.isNotBlank(poCountryOption)
```

Comment:
The same StringUtils.isNotBlank() check is getting duplicated in each "if" and "else if" condition. I would recommend adding some short-circuit logic in the event of a blank country option.

Also, StringUtils.equalsIgnoreCase() should already be null-safe, which can help with simplifying this.

Reply:
Adjusted the code.